src/__tests__/api.test.ts

```diff
@@ -35,42 +35,49 @@ jest.mock("https", () => {
 });
 
 jest.mock("https-proxy-agent", () => {
-  return jest
-    .fn()
-    .mockImplementation(
-      (opts: {
+  return jest.fn().mockImplementation(
+    (opts: {
+      protocol?: string;
+      host?: string;
+      port?: string;
+      pfx?: Buffer | string;
+      passphrase?: string;
+      headers?: {
+        "Proxy-Authorization"?: string;
+      };
+    }) => {
```

Comment:
The interface of the mock function should be as same of the original function.
Using `httpsProxyAgent.HttpsProxyAgentOptions` is more proper and fewer lines.

```suggestion
  return jest
    .fn()
    .mockImplementation((opts: httpsProxyAgent.HttpsProxyAgentOptions) => {
```